go/client/chat_common.go

```diff
@@ -4,13 +4,32 @@
 package client
 
 import (
+	"fmt"
+	"strings"
+
 	"golang.org/x/net/context"
 
 	"github.com/keybase/client/go/libkb"
 	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/teams"
 )
 
 func CheckUserOrTeamName(ctx context.Context, g *libkb.GlobalContext, name string) (*keybase1.UserOrTeamResult, error) {
+	userCli, err := GetUserClient(g)
+	if err != nil {
+		return nil, err
+	}
+	_, err = userCli.LoadUserByName(ctx, keybase1.LoadUserByNameArg{Username: name})
+	if err == nil {
+		ret := keybase1.UserOrTeamResult_USER
+		return &ret, nil
+	}
+	_, okNotFound := err.(libkb.NotFoundError)
+	_, okInvalidUsername := err.(libkb.BadUsernameError)
+	if !(okNotFound || okInvalidUsername || strings.HasPrefix(err.Error(), "bad keybase username")) {
```

Comment:
unsure why returned error `err` is `error.errorString` instead of `libkb.BadUsernameError`, but due to this, have to check error string